src/InnoSetup/IdfToolsSetup.iss

```diff
@@ -238,12 +238,16 @@ Source: "{#BUILD}\tools\xtensa-esp32s2-elf\*"; DestDir: "\\?\{app}\tools\xtensa-
 
 #ifndef DISABLE_TARGET_ESP32_C3
 Source: "{#BUILD}\tools\riscv32-esp-elf\*"; DestDir: "\\?\{app}\tools\riscv32-esp-elf"; Components: "{#COMPONENT_TARGET_ESP32_C3}"; Flags: recursesubdirs skipifsourcedoesntexist;
+Source: "{#BUILD}\tools\riscv32-esp-elf-gdb\*"; DestDir: "\\?\{app}\tools\riscv32-esp-elf-gdb"; Components: "{#COMPONENT_TARGET_ESP32_C3}"; Flags: recursesubdirs skipifsourcedoesntexist;
```

Comment:
Since there are also COMPONENT_TARGET_ESP32_C2 and COMPONENT_TARGET_ESP32_C6 now, both riscv32-esp-elf-gdb and riscv32-esp-elf should be installed when either of those components are selected.

riscv32-esp-elf should also be installed for ESP32-S2 and ESP32-S3 for ULP coprocessor support.

Reply:
Thank you @igrr for pointing out this issue. It seems that we might be able to make correction for mentioned case.

The different targets in different versions are stretching Inno Setup to it's maximum limit. Number of ifdef reached managable maximum. Features like component selection from dynamic list is the reason why we need espup, let's focus on this part of development in 2023 with @SergioGasquez.